dialog_lib/agents/abstract.py

```diff
@@ -175,4 +295,41 @@ def llm(self):
                 )
         return LLMChain(
             **chain_settings
-        )
\ No newline at end of file
+        )
+
+
+class AbstractLCELDialog(AbstractLCEL):
```

Comment:
why do we need an abstraction of an abstraction with LCEL here? It looks more like an rag example which used DialogRetriever, i.e., it'd live in another file like `dialog.py` where some `DialogAgent` (more coherent with the parent folder `agents`) or `DialogLLM` would inherent from `AbstractLCEL`. Since `AbstractLCEL` is abstract, would not need another abstract over it, seems more complicated to understand (there are 5 abstract classes already).

Reply:
We don't need it, I'm solving this right now!